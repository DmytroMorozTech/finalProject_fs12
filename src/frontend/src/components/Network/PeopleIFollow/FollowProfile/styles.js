import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  followProfile: {
    width: '20%',
    minWidth: '115px',
    borderRight: theme.border.simple,
    borderBottom: theme.border.simple
  },

  userInfo: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '225px',
    padding: theme.spacing(3)
  },

  link: {
    textDecoration: 'none'
  },

  largeAvatar: {
    width: theme.avatar.large,
    borderRadius: '50%'
  },

  name: {
    fontSize: theme.typography.h4.fontSize,
    fontWeight: theme.typography.h3.fontWeight,
    color: theme.palette.grey[700],
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(1),
    cursor: 'pointer'
  },

  smallLine: {
    width: '50px',
    height: '0.5px',
    border: '0',
    marginLeft: '0',
    marginBottom: theme.spacing(3),
    backgroundColor: theme.palette.grey[300]
  },

  line: {
    width: '100%',
    height: '0.5px',
    border: '0',
    margin: '0 auto',
    backgroundColor: theme.palette.grey[300]
  },

  follow: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: theme.spacing(2),
    cursor: 'pointer',
    fontSize: theme.typography.icons.medium.fontSize,
    color: theme.palette.grey[700],
    '&:hover': {
      backgroundColor: 'rgba(112, 181, 249, 0.2)'
    }
  },

  followTitle: {
    marginLeft: theme.spacing(1)
  }
}))
