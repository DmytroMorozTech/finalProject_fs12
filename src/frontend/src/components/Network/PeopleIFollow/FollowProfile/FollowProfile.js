import { Link } from 'react-router-dom'
import Image from '../../../../shared/Image/Image'
import { useState } from 'react'
import styles from './styles'
import Typography from '@material-ui/core/Typography'
import CheckIcon from '@material-ui/icons/Check'
import AddIcon from '@material-ui/icons/Add'

function FollowProfile (props) {
  const {id = 6, avatarUrl, fullName = 'Sheldon Cooper', workPlace = 'Scientist', numberFollowers = 50} = props.user
  const classes = styles()

  const linkToUserProfile = '/profile' + id

  const [follow, setFollow] = useState(true)

  const handleFollow = () => {
    follow ? setFollow(false) : setFollow(true)
  }

  return (
    <div className={classes.followProfile}>
      <div className={classes.userInfo}>
        <div>
          <Link to={linkToUserProfile} className={classes.link}>
            {/* <img alt='user`s avatar' src={avatar} className={classes.largeAvatar}/> */}
            <Image
              imageUrl={avatarUrl || 'linkedin/general/ghrchekikx3dnas6ivxm'}
              className={classes.largeAvatar}
              type={'profileAvatar'}
            />
          </Link>
          <Link to={linkToUserProfile} className={classes.link}>
            <Typography variant="h4" className={classes.name}>
              {fullName}
            </Typography>
            <Typography variant="h4" color="textSecondary">
              {workPlace}
            </Typography>
          </Link>
        </div>
        <div>
          <hr className={classes.smallLine}/>
          <Typography variant="h6">
            {numberFollowers} followers
          </Typography>
        </div>
      </div>
      <hr className={classes.line}/>
      <div className={classes.follow} onClick={handleFollow}>
        {follow
          ? <CheckIcon fontSize='inherit'/>
          : <AddIcon fontSize='inherit' color='primary'/>
        }
        {follow
          ? <Typography variant="h5" className={classes.followTitle}>Following</Typography>
          : <Typography variant="h5" color='primary' className={classes.followTitle}>Follow</Typography>
        }
      </div>
    </div>
  )
}

export default FollowProfile
