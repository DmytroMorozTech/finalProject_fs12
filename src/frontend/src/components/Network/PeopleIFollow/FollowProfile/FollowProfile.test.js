import { shallow } from 'enzyme'
import FollowProfile from './FollowProfile'
import * as redux from 'react-redux'
import React from 'react'

const user = {
  'id': 4,
  'createdDate': '2015-05-11T22:21:12.123123',
  'firstName': 'Everett',
  'lastName': 'Anderson',
  'fullName': 'Everett Anderson',
  'avatarPublicId': 'linkedin/avatars/i1bzhpqac3i1ld2urju8',
  'profileBgPublicId': '',
  'positionAndCompany': 'Project manager at HQSoftware',
  'email': 'test4@gmail.com',
  'isFollowedByActiveUser': true,
  'numberOfFollowers': 4
}

jest.mock('./styles', () => () => ({}))

const setUp = (props) => shallow(<FollowProfile {...props}/>)

describe('should render PeopleIFollowFollowers component', () => {
  let spyOnUseDispatch
  let mockDispatch

  let component
  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp({ user })
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
