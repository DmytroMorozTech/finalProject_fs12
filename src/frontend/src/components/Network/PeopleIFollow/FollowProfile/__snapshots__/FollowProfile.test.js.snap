// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PeopleIFollowFollowers component to match snapshot 1`] = `
<div>
  <div>
    <div>
      <Link
        to="/profiles/4"
      >
        <Image
          imageUrl="linkedin/avatars/i1bzhpqac3i1ld2urju8"
          type="profileAvatar"
        />
      </Link>
      <Link
        to="/profiles/4"
      >
        <WithStyles(ForwardRef(Typography))
          variant="h4"
        >
          Everett Anderson
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
          variant="h6"
        >
          Project manager at HQSoftware
        </WithStyles(ForwardRef(Typography))>
      </Link>
    </div>
    <div>
      <hr />
      <WithStyles(ForwardRef(Typography))
        variant="h6"
      >
        4
         followers
      </WithStyles(ForwardRef(Typography))>
    </div>
  </div>
  <hr />
  <div
    onClick={[Function]}
  >
    <Memo(ForwardRef(CheckIcon))
      fontSize="inherit"
    />
    <WithStyles(ForwardRef(Typography))
      variant="h5"
    >
      Following
    </WithStyles(ForwardRef(Typography))>
  </div>
</div>
`;
